app/lib/exchange-rate/providers/coingecko.ts

```diff
@@ -0,0 +1,55 @@
+import type {
+  ExchangeRateProvider,
+  GetRatesParams,
+  Rates,
+} from '~/lib/exchange-rate/providers/types';
+
+export class Coingecko implements ExchangeRateProvider {
+  supportedTickers: string[] = [
+    'BTC-USD',
+    'BTC-EUR',
+    'BTC-GBP',
+    'BTC-CAD',
+    'BTC-CHF',
+    'BTC-AUD',
+    'BTC-JPY',
+  ];
+
+  async getRates({ tickers, signal }: GetRatesParams): Promise<Rates> {
+    if (!tickers.length) {
+      throw new Error('No tickers provided');
+    }
+
+    tickers.forEach((ticker) => {
+      if (!this.supportedTickers.includes(ticker)) {
+        throw new Error(`Unsupported ticker: ${ticker}`);
+      }
+    });
+
+    const response = await fetch(
```

Comment:
we need to handle non ok responses as well. I didn't want to spend time on that. we have these options:

1. we handle it here
2. we create some thin fetch wrapper that does that
3. we instal some lib like ky or wretch 

I would maybe go with 3 and pick ky lib https://github.com/sindresorhus/ky. It has api similar to fetch and has nice fetures like timeout, retry etc. If we do that we can also enable timeout here the same way you had in your initial approach

Reply:
added in #237